process/process.go

```diff
@@ -39,25 +39,39 @@ const (
 )
 
 // convert pid to process object
-func NewProcess(id int) (p Process, err error) {
-	// read process status
+func NewProcess(id int) (Process, error) {
+	var (
+		p        Process
+		pid      int
+		ppid     int
+		pgrp     int
+		statByte []byte
+		encap    bool
+		err      error
+	)
```

Comment:
I didn't understand benefit of this format. Can you describe this format?

Reply:
Named returns are prone to errors, so it is highly encouraged not to use them.
Nevertheless you still need to declare your variables somewhere.
2 options: 
1) in line, where I mean by use of the :=
```
stat, err := someFunc()
if err != nil {return err}
```
this has some disadvantages:
- you have variable declarations "everywhere" in your code
- If same variable changes throughout the function you need to use = instead (so sometimes := then = ), so consistency
2) at the beginning of the function
- advantage:
- you have a direct overview of all variables used and there type
- you can use the shorthand 'if stat, err = someFunc(); err != nil {return err}'
- consist use of = instead of := / =
- disadvantage: long list of var declarations

However in this case here It can be shortened by assigning your pid directly to the struct elements and not using an extra assignment intermediate (would also use less memory) I'll address that in another PR